### PR TITLE
add UDF file format reader to support solutions add-on installation

### PR DESCRIPTION
**Pull request by vmanasiev**
## Short Description
An internal package to support installation of  VCD Solution Add-ons

## Description
In this package a reader is implemented that supports extracting files from solutions add-on ISO without unpacking/mounting it advance. 
## Motivation
A new feature is being implemented where the vcd client and terraform provider should support installation of the  VCD solution add-ons. Solution add-ons are packaged as an ISO media. To support larger files and avoid other issues with simple CDROM file format, the .iso file is packaged as  UDF filesystem (specification: http://www.osta.org/specs/pdf/udf260.pdf ).
udf internal package implements partially that specification and enables read. Specifically, the code allows only reading of top level objects within a .ISO file.

### CODE REVIEW

**Review by dataclouder — APPROVED** [review]

**Review by Didainius — APPROVED** [review]

**Review by lvirbalas — APPROVED** [review]
Thank you for your contribution. Approving in advance with a few requests below:

@vmanasiev or @dataclouder please add a few things:
1. Amend the PR description to state that this code is extracting only files from the top of the ISO tree.
2. Add top-level comment/description to the .go files (which are missing it) to explain the purpose and expectations of the code of a particular file.

**Comment by dataclouder** [comment]
> Thank you for your contribution. Approving in advance with a few requests below:
> 
> @vmanasiev or @dataclouder please add a few things:
> 
>     1. Amend the PR description to state that this code is extracting only files from the top of the ISO tree.
> 
>     2. Add top-level comment/description to the .go files (which are missing it) to explain the purpose and expectations of the code of a particular file.

I will take care of the code description after merging.

**Pull request merged**
